packages/pyright-internal/src/languageService/autoImporter.ts

```diff
@@ -158,9 +158,9 @@ export function buildModuleSymbolsMap(program: ProgramView, files: readonly Sour
                         continue;
                     }
 
-                    if (declaration.type === DeclarationType.Alias && isUserCode(file)) {
+                    if (declaration.type === DeclarationType.Alias && isUserCode(file) && !symbol.isInDunderAll()) {
                         // We don't include import alias in auto import
-                        // for workspace files.
+                        // for workspace files, unless they're in '__all__'
```

Comment:
i usually use explicit re-exports instead of `__all__`, and it looks like we can fix that too which will resolve #722

```ts
if (
    // We don't include import alias in auto import for workspace files.
    declaration.type === DeclarationType.Alias &&
    isUserCode(file) &&
    // unless they're in '__all__'
    !symbol.isInDunderAll() &&
    // or an alias with the same name (explicit re-export)
    (declaration.node.nodeType !== ParseNodeType.ImportFromAs ||
        declaration.node.d.alias?.d.value !== name)
) {
    continue;
}
```